### PR TITLE
drivers: gps: nrf9160_gps: emit started event

### PR DESCRIPTION
**Pull request by jhn-nordic**
Emit GPS_EVT_SEARCH_STARTED also when gps starts automatically
after position fix or timeout.

Jira: CIA-115

Signed-off-by: Jon Helge Nistad <jon.helge.nistad@nordicsemi.no>

### CODE REVIEW

**Review by jtguggedal — APPROVED** [review]

**Review by anangl — APPROVED** [review]

File: drivers/gps/nrf9160_gps/nrf9160_gps.c [hunk 1]
```diff
@@ -321,6 +323,14 @@ static void gps_thread(int dev_ptr)
 
 		switch (raw_gps_data.data_id) {
 		case NRF_GNSS_PVT_DATA_ID:
+			if (atomic_get(&drv_data->timeout_occurred) ||
+			   ((drv_data->current_cfg.nav_mode != GPS_NAV_MODE_CONTINUOUS) &&
```
Comment (simensrostad) [inline drivers/gps/nrf9160_gps/nrf9160_gps.c:327]: Alignment?

File: drivers/gps/nrf9160_gps/nrf9160_gps.c [hunk 2]
```diff
@@ -275,12 +277,12 @@ static void gps_thread(int dev_ptr)
...
 			/** If nrf_recv() blocks for more than one second, a fix
 			 *  has been obtained or the GPS has timed out. Submit
-			 *  a delayed timeout work every two seconds to make
+			 *  a delayed timeout work every five seconds to make
 			 *  sure the appropriate event is propagated upon
 			 *  a block.
 			 */
 			k_delayed_work_submit(&drv_data->timeout_work,
-					      K_SECONDS(2));
+					      K_SECONDS(5));
```
Comment (simensrostad) [inline drivers/gps/nrf9160_gps/nrf9160_gps.c:285]: Why the increase to five seconds here?

File: drivers/gps/nrf9160_gps/nrf9160_gps.c [hunk 3]
```diff
@@ -685,6 +695,7 @@ static int start(const struct device *dev, struct gps_config *cfg)
 	}
 
 	atomic_set(&drv_data->is_active, 1);
+	atomic_set(&drv_data->timeout_occurred, 0);
```
Comment (simensrostad) [inline drivers/gps/nrf9160_gps/nrf9160_gps.c:698]: Does it make sense to provide a common cleanup function for atomic variables?

**Review by simensrostad — COMMENTED** [review]
Looks good, just a few comments here and there.

File: drivers/gps/nrf9160_gps/nrf9160_gps.c (hunk 2 above)
Comment (jhn-nordic) [reply drivers/gps/nrf9160_gps/nrf9160_gps.c:285]: 2 is not sufficient. Saw several occasions where timeout event was sent. Those issues went away when increasing timeout.

File: drivers/gps/nrf9160_gps/nrf9160_gps.c (hunk 3 above)
Comment (jhn-nordic) [reply drivers/gps/nrf9160_gps/nrf9160_gps.c:698]: They are not set to the same value always. Here it is 1 and 0. In the next place it is 0 and 0.

File: drivers/gps/nrf9160_gps/nrf9160_gps.c (hunk 1 above)
Comment (jhn-nordic) [reply drivers/gps/nrf9160_gps/nrf9160_gps.c]: fixed

**Review by simensrostad — APPROVED** [review]
LGMT

**Review by shantha-14 — APPROVED** [review]

**Pull request merged**